.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: Main testing workflow
+on: [pull_request]
+
+env:
+
+  # name of docker image
+  image: cred-action
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Actions Repository
+      uses: actions/checkout@v2
+    - name: Build action container
+      env:
+        image: '${{ env.image }}'
+      run: docker build -t "${image}" .
+
+    # These envars are what would be set by the user in the workflow
+    - name: Test Action with Container
+      env:
+
+        # project-file
+        INPUT_PROJECT_FILE: .github/github.json
+
+        # project identifier
+        INPUT_PROJECT: '@sourcecred'
+
+        # branch-against
+        INPUT_BRANCH_AGAINST: "master"
+
+        # scores-json
+        INPUT_SCORES_JSON: scores.json
+
+        # weights
+        INPUT_WEIGHTS: .github/weights.json
+
+        # target
+        INPUT_TARGET: docs
+
+        # automated (not relevant since test run is set)
+        INPUT_AUTOMATED: false
+        INPUT_TEST_RUN: true
+
+        # github workspace would normally be honored in container
+        GITHUB_WORKSPACE: /github/workspace
+
+        image: '${{ env.image }}'
+        token: ${{ secrets.GITHUB_TOKEN }}
+        SOURCECRED_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      run: docker run --workdir /github/workspace --rm -e token -e INPUT_PROJECT_FILE -e INPUT_TARGET -e INPUT_PROJECT -e INPUT_AUTOMATED -e INPUT_TEST_RUN -e INPUT_BRANCH_AGAINST -e INPUT_SCORES_JSON -e INPUT_WEIGHTS -e SOURCECRED_GITHUB_TOKEN -e GITHUB_TOKEN -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "${PWD}/":"/github/workspace" "${image}"
```

Comment:
So, I poked around at the docs and looked at alternative tools.
We can avoid this `docker run` style emulation by using a relative path for the action reference instead.

https://github.com/vsoch/cred-action/blob/3ecdb0d8fb37a8e2d4ba954a96a50ad38a676fc4/.github/workflows/integration.yml#L16-L26

The `uses: ./` there will actually run the locally checked out `./action.yml`
rather than a remote one like `owner/repo@version`

See https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow

Reply:
I've also tested running this action locally using https://github.com/nektos/act, but unfortunately has a bug here when evaluating the `action.yml`'s parameters. Not renaming them from `project-file` to `INPUT_PROJECT_FILE`, checking for required parameters, or setting default values.

The point is though, this _will_ run on the real github actions :]

Reply:
Yes, I think it must be that it doesn't convert the action variables to the environment in the action.yml? Since project file has a dash that is converted to `INPUT_PROJECT-FILE` by actions, so instead I set the input.project-file to be `INPUT_PROJECT_FILE`. I really like the idea of doing uses ./, let's try that out!